velox/functions/prestosql/aggregates/tests/MapUnionAggregationTest.cpp

```diff
@@ -345,5 +345,77 @@ TEST_F(MapUnionTest, unknownKeysAndValues) {
       AssertQueryBuilder(plan).copyResults(pool()), "map key cannot be null");
 }
 
+TEST_F(MapUnionTest, nans) {
+  // Verify that NaNs with different binary representations are considered equal
+  // and deduplicated when used as keys in the output map.
+  static const auto kNaN = std::numeric_limits<double>::quiet_NaN();
+  static const auto kSNaN = std::numeric_limits<double>::signaling_NaN();
+
+  // Global Aggregation, Primitive type
+  auto data = makeRowVector(
+      {makeMapVectorFromJson<double, int32_t>(
+           {"{1: 1}", "{2: 2}", "{NaN: 4}", "{3: 3}", "{NaN: 5}", "{NaN: 6}"}),
+       makeFlatVector<int32_t>({1, 1, 1, 2, 2, 2})});
+
+  auto expectedResult = makeRowVector({
+      makeMapVectorFromJson<double, int32_t>({
+          "{ 1: 1, 2: 2, 3: 3, NaN: 4}",
+      }),
+  });
+
+  testAggregations({data}, {}, {"map_union(c0)"}, {expectedResult});
```

Comment:
curious ; are we always guaranteed that map_union will return the first key:value (i.e NaN:4) ?

Reply:
It returns the first key encountered. Similar to my previous comment, this should be deterministic due to the small single vector input.